packages/codegen/src/index.ts

```diff
@@ -138,7 +140,16 @@ if (require.main === module) {
   if (Object.fromEntries === undefined) {
     throw new Error("Joist requires Node v12.4.0+");
   }
-  main().catch((err) => {
+  main().then(() => {
+    if (
+        // strict mode + warnings or greater
+        (process.argv.includes("--strict") && loggerMaxWarningLevelHit >= LOG_LEVELS.warn) ||
```

Comment:
Nice!

Per the immediate-fks thing, right, I'm kinda purposefully/lazily using that codegen failure as a poor-mans test harness, vs. making a truly isolated unit test that faked everything out and could trigger/cover the same behavior (which is definitely the most kosher approach, long-term)...

Dunno, maybe we could add a `--never-fail` / `--ignore-errors` / `--always-exit-code-zero` type check here, that is really only intended for that one immediate-fks test `package.json` to use... I think that'd work? And not to too bad...

Reply:
Pushed with --always-exit-code-zero as i think it's the clearest.